database/select_account.go

```diff
@@ -16,11 +16,20 @@ func SelectAccount(db *sql.DB, id string) (*model.Account, error) {
 		FROM account
 		WHERE id = ? FOR UPDATE;`
 
+	var err error
 	account := &model.Account{}
-	err := db.QueryRow(selectAccountStmt, id).Scan(&account.ID,
-		&account.Username,
-		&account.Password,
-		&account.Plan)
+	if tx == nil {
+		err = db.QueryRow(selectAccountStmt, id).Scan(&account.ID,
+			&account.Username,
+			&account.Password,
+			&account.Plan)
+	} else {
+		err = tx.QueryRow(selectAccountStmt, id).Scan(&account.ID,
+			&account.Username,
+			&account.Password,
+			&account.Plan)
+	}
```

Comment:
Instead of doing this, you can make this function just accept and interface:

```
type query interface {
   QueryRow(query string, args ...interface{}) *sq.Row
}
```

Then you don't have to pass `nil` when you call your functions.